Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -155,6 +155,155 @@ static int CompareGameListItems(const GameListItem* iso1, const GameListItem* is
   return 0;
 }
 
+static std::unordered_map<std::string, std::string> LoadCustomTitles()
+{
+  // Load custom game titles from titles.txt
+  // http://www.gametdb.com/Wii/Downloads
+  const std::string& load_directory = File::GetUserPath(D_LOAD_IDX);
+
+  std::ifstream titlestxt;
+  OpenFStream(titlestxt, load_directory + "titles.txt", std::ios::in);
+
+  if (!titlestxt.is_open())
+    OpenFStream(titlestxt, load_directory + "wiitdb.txt", std::ios::in);
+
+  if (!titlestxt.is_open())
+    return {};
+
+  std::unordered_map<std::string, std::string> custom_titles;
+
+  std::string line;
+  while (!titlestxt.eof() && std::getline(titlestxt, line))
+  {
+    const size_t equals_index = line.find('=');
+    if (equals_index != std::string::npos)
+    {
+      custom_titles.emplace(StripSpaces(line.substr(0, equals_index)),
+                            StripSpaces(line.substr(equals_index + 1)));
+    }
+  }
+
+  return custom_titles;
+}
+
+static std::vector<std::string> GetFileSearchExtensions()
+{
+  std::vector<std::string> extensions;
+
+  if (SConfig::GetInstance().m_ListGC)
+  {
+    extensions.push_back(".gcm");
+    extensions.push_back(".tgc");
+  }
+
+  if (SConfig::GetInstance().m_ListWii || SConfig::GetInstance().m_ListGC)
+  {
+    extensions.push_back(".iso");
+    extensions.push_back(".ciso");
+    extensions.push_back(".gcz");
+    extensions.push_back(".wbfs");
+  }
+
+  if (SConfig::GetInstance().m_ListWad)
+    extensions.push_back(".wad");
+
+  if (SConfig::GetInstance().m_ListElfDol)
+  {
+    extensions.push_back(".dol");
+    extensions.push_back(".elf");
+  }
+
+  return extensions;
+}
+
+static bool ShouldDisplayGameListItem(const GameListItem& item)
+{
+  bool list = true;
+
+  switch (item.GetPlatform())
+  {
+  case DiscIO::Platform::WII_DISC:
+    if (!SConfig::GetInstance().m_ListWii)
```

Comment:
I've no idea if it was like that for any particular reason. cc @JosJuice 

Technically from what I gather, it can just be:

```cpp
static bool ShouldDisplayGameListItem(const GameListItem& item)
{
  const bool show_platform = [&item] {
    switch (item.GetPlatform())
    {
    case DiscIO::Platform::GAMECUBE_DISC:
      return SConfig::GetInstance().m_ListGC;
    case DiscIO::Platform::WII_DISC:
      return SConfig::GetInstance().m_ListWii;
    case DiscIO::Platform::WII_WAD:
      return SConfig::GetInstance().m_ListWad;
    case DiscIO::Platform::ELF_DOL:
      return SConfig::GetInstance().m_ListElfDol;
    default:
      return false;
    }
  }();

  if (!show_platform)
    return false;

  switch (item.GetCountry())
  {
  case DiscIO::Country::COUNTRY_AUSTRALIA:
    return SConfig::GetInstance().m_ListAustralia;
  case DiscIO::Country::COUNTRY_EUROPE:
    return SConfig::GetInstance().m_ListPal;
  case DiscIO::Country::COUNTRY_FRANCE:
    return SConfig::GetInstance().m_ListFrance;
  case DiscIO::Country::COUNTRY_GERMANY:
    return SConfig::GetInstance().m_ListGermany;
  case DiscIO::Country::COUNTRY_ITALY:
    return SConfig::GetInstance().m_ListItaly;
  case DiscIO::Country::COUNTRY_JAPAN:
    return SConfig::GetInstance().m_ListJap;
  case DiscIO::Country::COUNTRY_KOREA:
    return SConfig::GetInstance().m_ListKorea;
  case DiscIO::Country::COUNTRY_NETHERLANDS:
    return SConfig::GetInstance().m_ListNetherlands;
  case DiscIO::Country::COUNTRY_RUSSIA:
    return SConfig::GetInstance().m_ListRussia;
  case DiscIO::Country::COUNTRY_SPAIN:
    return SConfig::GetInstance().m_ListSpain;
  case DiscIO::Country::COUNTRY_TAIWAN:
    return SConfig::GetInstance().m_ListTaiwan;
  case DiscIO::Country::COUNTRY_USA:
    return SConfig::GetInstance().m_ListUsa;
  case DiscIO::Country::COUNTRY_WORLD:
    return SConfig::GetInstance().m_ListWorld;
  case DiscIO::Country::COUNTRY_UNKNOWN:
  default:
    return SConfig::GetInstance().m_ListUnknown;
  }
}
```